### DISCUSSION TITLE
Transfer of information between channels

### DISCUSSION THREAD

**Discussion by bkojin90** [nextgenhealthcare/connect#5771] [Q&A]
Hi everyone,

I am new to the world of Mirth Connect and am just trying to figure out some basic concepts. I'd read and searched a lot for the answers but am no closer to solving these questions it seems!

How do I best send information from one channel ('Channel A') to another channel ("Channel B').
What I want is for Channel A in this case to read a cell from an SQL database and then pass that information to be used in Channel B. Where Channel B writes to the SQL table based on the information from Channel A.

I've got it to read and write from my SQL database but I cannot figure out how to transfer a simple piece of information between channels. I assumed it might be channel metadata or something but I must not be getting the syntax correct.

Would any kind soul be willing to lend their expertise and advise on what I should be doing?

Thank you so much!
Bin

**Comment by rjkroll** [comment]
There are various ways of passing information between channels.    You can just set all your data that you need to channelMap variables, then use them in your Destination Channel Writer to send this data to Channel B.    You can also pull your SQL data, map it to whatever variables, then use a Javascript transformer and use the built-in router functions to route the data directly to the other channel.

In your case, though, it sounds like you only need one channel.   In your Source, you connect to Database A, run your query, get your data, map to variables.   In your Destination, just use a Database connection and point to Database B.   Then, you can just build an Insert statement for Database B in your Destination and pass in the variables from your Source (that you pulled from Database A)

**Comment by pacmano1** [comment]
@bkojin90 , please post screenshots of what you tried.

Sending from one channel to another can be done using a destination of type "channel writer" or using a `router.routeMessage` statement in nearly anywhere in mirth.  The contents of the message sent to the next channel can be the orginal message you recevied or that message transformed or really anything with our without message metadata (you will see that section when you select "channel writer" in a destiantion).

@rjkroll mentioned "mapping to variables", that isn't required per se if you do not change (i.e. "transform") the message when it exits the source and starts processing destinations since the original message can be parsed as needed in a destination without the use of variables.    He's not wrong though, it's just a question of how someone prefers to work with Mirth.

If you are brand new to mirth, you should consider some formal training to get you up to speed.   You might also consider books like https://github.com/nextgenhealthcare/connect/discussions/5771 and books from https://mirthconnect.shamilpublishing.com/.

**Reply by bkojin90** [reply]
ah okay, yes that makes sense. Thank you so much!

The other odd problem I seem to be having is with the SQL syntax, where trying to run the following SQL code in a database writer:

INSERT INTO test_123_2023-05-01_1_testing (client_name)
VALUES (${message.rawData)}

this doesn't seem to work but I was wondering if it was to do with the '-' and '_' in the table title?

Any help would be wonderful :)

**Reply by pacmano1** [reply]
Your syntax is not correct.  Please learn to use code blocks on Github.   

```sql
INSERT INTO test_123_2023-05-01_1_testing (client_name)
VALUES (${message.rawData})
```

I suspect you do not wnat message.rawData though rather some value from the message.

**Reply by bkojin90** [reply]
My apologies pacmano1, I am new to this. Thank you for your help.

For testing purposes I am happy to send the raw data. It is just odd that it gives errors like 'the column index is out of range'. When I am definitely only trying to put 1 thing into 1 column.

**Reply by bkojin90** [reply]
oh and thank you for the book recommendations! I will be reading those for sure in the coming weeks.

**Reply by bkojin90** [reply]
I switched around the '}' and the ')' in the code-- that was a typo I made here, my apologies

**Reply by bkojin90** [reply]
ah I think mirth connect doesn't like '-' or '_' characters at all anywhere!
